app/controllers/causes_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CausesController < ApplicationController
+  before_action :authenticate_user!, only: %i{create destroy}
+  before_action :correct_user, only: %i{destroy}
+
+  def new
+    @cause = Cause.new
```

Comment:
```suggestion
    @cause = current_user.causes.build
```
xong bỏ .merge({user_id: params[:user_id]}) ở hàm cause_params nhé